### PR TITLE
chore: add more logging for failures to get label metadata

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Need better error messages. https://cloud-native.slack.com/archives/C01TSERG0KZ/p1687308760074909

### CODE REVIEW

**Review by jsoref — APPROVED** [review]
Sounds good to me

File: util/argo/resource_tracking.go [hunk 1]
```diff
@@ -151,18 +156,26 @@ func (rt *resourceTracking) SetAppInstance(un *unstructured.Unstructured, key, v
 		if len(val) > LabelMaxLength {
 			val = val[:LabelMaxLength]
 		}
-		return argokube.SetAppInstanceLabel(un, key, val)
+		err = argokube.SetAppInstanceLabel(un, key, val)
+		if err != nil {
+			return fmt.Errorf("failed to set app instance label: %w", err)
```
Comment (jsoref) [inline util/argo/resource_tracking.go:161]: Actually, do you know the name of the app?

Comment (crenshaw-dev) [reply util/argo/resource_tracking.go:161]: I'll add that to some more places where this is logged...

**Review by ishitasequeira — APPROVED** [review]
Looks good!!

**Review by jannfis — APPROVED** [review]
LGTM

**Comment by crenshaw-dev** [comment]
Cherry picking because I think there might be a regression in 2.8 related to handling of null labels and annotations in manifests. Hoping the additional logs will help us debug.

**Pull request merged**